src/viewer.js

```diff
@@ -1227,7 +1120,8 @@ class VolumeImageViewer {
         loaderParams: {
           pyramid: pyramid,
           client: this[_options].client,
-          channel: opticalPathIdentifier
+          channel: opticalPathIdentifier,
+          iccProfiles: _getIccProfiles(pyramid, this[_options].client)
```

Comment:
Unfortunately there are two major issue with the fetching of the icc profiles when we initialize the wasm module for the ColorTransformer. The first in terms of performances, the second is technical: 
1) if you have N web workers (for each web worker we need to load the wasm module), you would need to fecth the icc profiles N times, wasting bandwith. 
2) we can't pass objects (i.e.,`client`) to webworkers. Therefore we actaully can't fetch the icc profiles when we initialize the wasm module for the ColorTransformer.

So I moved the fetching to the constructor (so it happens only once) and then mapped this with ColorManager transfomers by SOPInstanceUID. If you have a better idea, please let me know.

Reply:
Good point!

I think we should then instantiate the decoders and transformers in `render()` rather than in `constructor()`.

Reply:
the decoders and transformers are already instantiated at render time (i.e. at the first time a decoding task is requested inside a web worker object, please see https://github.com/herrmannlab/dicom-microscopy-viewer/pull/76#issuecomment-1126027888). 

While currently the ICC profiles (the buffer) are fetched at constructor level and then passed to the webworkers with the async function _createTileLoadFunction, which then pass the ICC profiles buffer to the webworker task in _decodeAndTransformFrame. 

Ideally we woul pass a `DICOMwebClient` instance into the webworker and then fetch the ICC profiles when instantiating the transfomer WASM module in each worker (this would me to fetch the ICC profiles for each worker, wasting bandwidth, but it would put the ICC profiles fecthing at rendering time). However, the main problem is that we can't pass objects (i.e. a DICOMwebClient instance) to web workers. So the only way is to pass to the webworkers, it is to pass directly the already fetched ICC profiles (the buffer) to the webworkers. 

Finally, I don't see any way to fetch them in another place which is not the constructor, because the tile loader function (`_createTileLoadFunction`) is called in the constructor (and we pass the informations regarding the tiles to the webworkers by the `_createTileLoadFunction`).

Reply:
We really shouldn't do any data fetching in the constructor. That is very much against the current design of the library.

> Ideally we woul pass a DICOMwebClient instance into the webworker and then fetch the ICC profiles when instantiating the transfomer WASM module in each worker (this would imply to fetch the ICC profiles for each worker, wasting bandwidth, but it would put the ICC profiles fecthing at rendering time). However, the main problem is that we can't pass objects (i.e. a DICOMwebClient instance) to web workers. So the only way to pass the ICC profiles to the webworkers, it is to pass directly the already fetched ICC profiles (the buffer) to the webworkers.

That's fine. We can keep passing the value of ICC Profile to the workers. The main question is then where and when the workers are initialized. I tend argue that this also only happen when `render()` is called.

> I don't see any way to fetch them in another place which is not the constructor, because the tile loader function (_createTileLoadFunction) is called in the constructor (and we pass the informations regarding the tiles to the webworkers by the _createTileLoadFunction).

Wouldn't it be possible to create the tile load function and set it on the OpenLayers source in the `render()` method? If necessary, we could set a dummy tile load function in the constructor and update it once `render()` is called.

Reply:
you are 100% right, we can move the fetching into the render. Done in https://github.com/herrmannlab/dicom-microscopy-viewer/pull/76/commits/b36f053e618a594e34da0b4c36cf6ac92dea5764